api/src/adapters/aws/api_gateway_adapter.py

```diff
@@ -0,0 +1,167 @@
+    enabled: bool = Field(alias="enabled", default=True)
+    stage_keys: list[str] = Field(alias="stageKeys", default_factory=list)
+    tags: dict[str, str] = Field(alias="tags", default_factory=dict)
+
+
+class ApiGatewayConfig(PydanticBaseEnvConfig):
+    """Configuration for AWS API Gateway integration"""
+
+    # Usage plan ID for newly created API keys (typically the public usage plan)
+    default_usage_plan_id: str | None = Field(
+        default=None, alias="API_GATEWAY_DEFAULT_USAGE_PLAN_ID"
+    )
+
+
+def get_boto_api_gateway_client(session: boto3.Session | None = None) -> botocore.client.BaseClient:
+    """Get a boto3 API Gateway client"""
+    if session is None:
+        session = get_boto_session()
+
+    return session.client("apigateway")
+
+
+def import_api_key(
+    api_key: str,
+    name: str,
+    description: str | None = None,
+    enabled: bool = True,
+    usage_plan_id: str | None = None,
+    api_gateway_client: botocore.client.BaseClient | None = None,
+) -> ApiKeyImportResponse:
+    """
+    Import an API key into AWS API Gateway using CSV format.
+
+    This function uses AWS API Gateway's CSV import functionality to import an API key
+    and associate it with a usage plan in a single operation, eliminating the need for
+    separate usage plan association calls.
+
+    Args:
+        api_key: The API key value to import (must be 20-128 characters)
+        name: Name for the API key (cannot exceed 1024 characters)
+        description: Optional description for the API key
+        enabled: Whether the API key should be enabled (default: True)
+        usage_plan_id: Optional usage plan ID to associate the key with during import
+        api_gateway_client: Optional pre-configured API Gateway client
+
+    Returns:
+        ApiKeyImportResponse with the imported key details
+    """
+
+    if is_local_aws():
+        return _handle_mock_import_response(api_key, name, description, enabled, usage_plan_id)
+
+    if api_gateway_client is None:
+        api_gateway_client = get_boto_api_gateway_client()
+
+    # Format the API key data as CSV for import
+    # AWS API Gateway expects CSV format: name,key,description,enabled,usageplanIds
+    # Header row is optional but improves readability and maintainability
+    usage_plan_ids_str = f'"{usage_plan_id}"' if usage_plan_id else ""
+    header = "name,key,description,enabled,usageplanIds"
+    data_row = f"{name},{api_key},{description or ''},{'true' if enabled else 'false'},{usage_plan_ids_str}"
+    csv_data = f"{header}\n{data_row}"
+
+    try:
+        response = api_gateway_client.import_api_keys(
+            body=csv_data.encode("utf-8"), format="csv", failOnWarnings=True
+        )
+    except ClientError:
+        logger.exception("Error importing API key to AWS API Gateway")
+        raise
+    except Exception as e:
+        logger.exception("Unexpected error importing API key", extra={"error": str(e)})
+        raise
```

Comment:
logger.exception adds the error automatically. Same with the except a bit below here
```suggestion
    except Exception:
        logger.exception("Unexpected error importing API key")
        raise
```